### PR TITLE
Separate RBAC into sub-directories to support RHOAM and other layered products

### PR DESCRIPTION
**Pull request by mciccone**
CSSRE needs to support multiple layered products and not just RHMI. The config changes in this PR move the contents of the /deploy/layered-sre-authorization directory into a  new sub-dir called RHMI (preserving CSSRE access functionality to all current RHMI 2.x clusters). 
We've also added another new sub-dir called RHOAM and duplicated the config from RHMI but altered it to support CSSRE access to upcoming layered product (RHOAM) clusters. 

fixes [CSSRE-1384](https://issues.redhat.com/browse/CSSRE-1384)

### CODE REVIEW

File: deploy/layered-sre-authorization/rhmi/02-layered-cs-sre-admin.SubjectPermission.yaml [hunk 1]
```diff
@@ -10,7 +10,7 @@ spec:
   permissions:
   - allowFirst: true
     clusterRoleName: layered-cs-sre-admin-project
-    namespacesAllowedRegex: .*
+    namespacesAllowedRegex: (^redhat-rhmi$|^redhat-rhmi-.*)
     namespacesDeniedRegex: (^kube$|^kube-.*|^openshift$|^openshift-.*|^default$)
```
Comment (sclarkso) [inline deploy/layered-sre-authorization/rhmi/02-layered-cs-sre-admin.SubjectPermission.yaml:14]: Based on my understanding, we don't need an allow and a deny.  see: https://github.com/openshift/managed-cluster-config/pull/545/files.  We will also need rhoam namespaces: "(^redhat-rhmi-.*|^redhat-rhoam-.*)"

Comment (Nanyte25) [reply deploy/layered-sre-authorization/rhmi/02-layered-cs-sre-admin.SubjectPermission.yaml:14]: The namespace prefix Regix should be `(^redhat-rhmi$|^redhat-rhmi-.*)` and `(^redhat-rhoam$|^redhat-rhoam-.*)`. We should explicitly deny all first and then allow, this is best practice here.

**Comment by jewzaam** [comment]
Need to prepare for backplane for CSSRE, see https://github.com/openshift/managed-cluster-config/tree/master/deploy/backplane files with "cssre" in the name.

File: deploy/layered-sre-authorization/rhmi/02-layered-cs-sre-admin.SubjectPermission.yaml (hunk 1 above)
Comment (alanmoran) [reply deploy/layered-sre-authorization/rhmi/02-layered-cs-sre-admin.SubjectPermission.yaml:14]: We can't do this in RHMI. It must remain as a `.*`. RHMI has components that dynamically creates namespaces based on what the customer creates. (In fuse AFAIK).

File: deploy/layered-sre-authorization/rhoam/01-rhoam-cs-sre-admin-cluster.ClusterRole.yaml [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+  - update
+  - get
+# CS SRE can interact with RHMI ClusterResourceQuotas (Only for LA)
+- apiGroups:
+  - quota.openshift.io
+  resources:
+  - clusterresourcequotas/status
+  - clusterresourcequotas
+  verbs:
+  - '*'
```
Comment (alanmoran) [inline deploy/layered-sre-authorization/rhoam/01-rhoam-cs-sre-admin-cluster.ClusterRole.yaml:61]: Needs to be removed. This was for RHMI LA only.

File: deploy/layered-sre-authorization/rhoam/config.yaml [hunk 3]
```diff
@@ -0,0 +1,6 @@
+deploymentMode: "SelectorSyncSet"
+selectorSyncSet:
+  matchLabels:
+      api.openshift.com/addon-rhoam-operator: "true"
+      api.openshift.com/addon-rhoam-operator-internal: "true"
```
Comment (alanmoran) [inline deploy/layered-sre-authorization/rhoam/config.yaml:5]: Is addon-rhoam-operator-internal a thing? I thought there was only one?

**Review by alanmoran — CHANGES_REQUESTED** [review]

File: deploy/layered-sre-authorization/rhoam/01-rhoam-cs-sre-admin-cluster.ClusterRole.yaml (hunk 2 above)
Comment (mciccone) [reply deploy/layered-sre-authorization/rhoam/01-rhoam-cs-sre-admin-cluster.ClusterRole.yaml:61]: Ok removing lines 54-61

File: deploy/layered-sre-authorization/rhoam/config.yaml (hunk 3 above)
Comment (mciccone) [reply deploy/layered-sre-authorization/rhoam/config.yaml:5]: I'm working to verify that operator-internal does not exist in RHOAM

Comment (alanmoran) [reply deploy/layered-sre-authorization/rhoam/config.yaml:5]: This won't exist for GA. Safe to remove and "matchLabelsApplyMode"

**Comment by mciccone** [comment]
> Need to prepare for backplane for CSSRE, see https://github.com/openshift/managed-cluster-config/tree/master/deploy/backplane files with "cssre" in the name.

@sclarkso modified the backplane files for cssre and we believe these change should suffice for now. We will review these changes again as the backplane service gets closer to release.

File: .gitignore [hunk 4]
```diff
@@ -1,2 +1,3 @@
 tmp_exporters/
 generated/
+.DS_Store
```
Comment (RiRa12621) [inline .gitignore:3]: Everyone should have that globally. No need for that here

**Review by RiRa12621 — CHANGES_REQUESTED** [review]

**Comment by RiRa12621** [comment]
Additionally I think you missing the generated files.

**Comment by mciccone** [comment]
> The following users are mentioned in OWNERS file(s) but are untrusted for the following reasons. One way to make the user trusted is to add them as members of the openshift org. You can then trigger verification by writing `/verify-owners` in a comment.
> 
> * hlipsig
>   
>   * User is not a member of the org. User is not a collaborator. Satisfy at least one of these conditions to make the user trusted.

**Comment by rogbas** [comment]
I don't see the template files updated in this PR, can you please refresh it and add them?

**Comment by cblecker** [comment]
/lgtm
/approve

I did a comparison to the current layered-sre-authorization files for RHMI, and the RHOAM ones look good (identical, except for the name in most cases).

**Comment by mrbarge** [comment]
I'll note that the introduction of the new group "rhoam-cs-sre-admins", this probably needs to be added to the user groups that are authorized/validated by `managed-cluster-validating-webhooks`. I've created OSD-5915 to track that.

**Pull request merged**